src/App.tsx

```diff
@@ -7,39 +7,88 @@ import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
-import { Loader } from './components/Loader';
+import { Loader } from './components/Loader/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { ErrMessage } from './types/ErrMessage';
+import { getPosts, getUsers } from './api';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [errorMess, setErrorMess] = useState<ErrMessage>(ErrMessage.Empty);
+  const [isLoadingPosts, setIsLoadingPosts] = useState<boolean>(false);
+  const [posts, setPosts] = useState<Post[] | null>(null);
```

Comment:
Consider initializing your posts state to an empty array instead of null. It's a good practice to have a consistent type for your state.